### PR TITLE
bpo-31792: Restore os.environ in test_buffer when import numpy.

### PR DESCRIPTION
**Pull request by serhiy-storchaka**
https://bugs.python.org/issue31792

### CODE REVIEW

**Review by skrah — APPROVED** [review]

File: Lib/test/test_buffer.py [hunk 1]
```diff
@@ -37,7 +37,8 @@
     ctypes = None
 
 try:
-    with warnings.catch_warnings():
+    with support.EnvironmentVarGuard() as os.environ, \
```
Comment (vstinner) [inline Lib/test/test_buffer.py:40]: Does it restore os.environ to its original value after the with block? Do we need the "as" part?

Comment (serhiy-storchaka) [reply Lib/test/test_buffer.py:40]: Yes, it restores os.environ to its original value after the with block. EnvironmentVarGuard() just creates a proxy to os.environ, it doesn't touch os.environ, and it doesn't track direct modifications of os.environ. We need the "as" part for making all changes done through the proxy.

Comment (vstinner) [reply Lib/test/test_buffer.py:40]: Oh, EnvironmentVarGuard doesn't set itself as os.environ, but its .__exit__() method restores os.environ to the previous object. I see.

File: Lib/test/test_buffer.py [hunk 2]
```diff
@@ -17,7 +17,7 @@
 from itertools import permutations, product
 from random import randrange, sample, choice
 import warnings
-import sys, array, io
+import sys, array, io, os
```
Comment (vstinner) [inline Lib/test/test_buffer.py:20]: PEP 8: please write one import per line. Reorder maybe also all imports (sort them all).

Comment (skrah) [reply Lib/test/test_buffer.py:20]: I'm maintaining this code and I'm quite happy with multiple imports per line.

Comment (serhiy-storchaka) [reply Lib/test/test_buffer.py:20]: Initially I rewrote this in multiple lines. But then I seen that it required reordering virtually every line. Taking into account that changes in this PR are not strongly necessary, I prefer to do with minimal changes.

**Pull request merged**